File: source/addonManager51/Views/uploadClass.vb
Imports Contensive.BaseClasses

Namespace Contensive.Addons.AddonManager51
    '
    Public Class UploadClass
        Inherits AddonBaseClass
        '
        ' -- injected objects -- do not dispose
        Private cp As CPBaseClass
        ' 
        ' -- class scope
        Private Const RequestNameButton As String = "button"
        Private Const ButtonCancel As String = " Cancel "
        Private Const ButtonOK As String = " OK "
        '
        '====================================================================================================
        '
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Try
                Me.cp = CP
                Return getUpload()
            Catch ex As Exception
                CP.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
        '
        '====================================================================================================
        '
        Private Function getUpload() As String
            Dim returnResult As String = ""
            Try
                Dim Button As String = cp.Doc.GetText(RequestNameButton)
                If Button = ButtonCancel Then
                    '
                    ' ----- redirect back to the root
                    Call cp.Response.Redirect(cp.Site.GetText("adminUrl"))
                Else
                    '
                    ' -- create form
                    Dim form As New PortalFramework.FormNameValueRowsClass With {
                        .title = "Upload Collection",
                        .body = cp.Html.p("Use this form to upload an add-on collection. If the GUID of the add-on matches one already installed on this server, it will be updated. If the GUID is new, it will be added."),
                        .description = cp.Html.p("Upload a collection zip file to install the collection on this site. ")
                    }
                    If Not cp.User.IsAdmin() Then
                        '
                        ' -- Put up error message
                        form.body &= cp.Html.p("You must be an administrator to use this tool.")
                    Else
                        If (cp.Site.GetText("buildVersion") < cp.Version) Then
                            '
                            ' -- database needs to be upgraded
                            form.description &= cp.Html.p("WARNING: The site database needs to be upgraded. You should do this before installing addon collections.")
                        End If
                        '
                        ' -- Upload tool
                        If (Button = ButtonOK) Then
                            '
                            ' -- handle upload
                            If cp.User.IsDeveloper And cp.Doc.GetBoolean("InstallCore") Then
                                '
                                ' -- Reinstall core collection
                                Call cp.Content.Delete("Add-on Collections", "ccguid='{8DAABAE6-8E45-4CEE-A42C-B02D180E799B}'")
                                '
                                ' -- use v5 method
                                Dim ErrorMessage As String = ""
                                If (Not InstallController.installCollectionFromLibrary(cp, "{8DAABAE6-8E45-4CEE-A42C-B02D180E799B}", ErrorMessage)) Then
                                    form.description &= cp.Html.p("ERROR: " & ErrorMessage)
                                End If
                            End If
                            Dim uploadFilename As String = cp.Doc.GetText(rnUploadCollectionFile)
                            If (Not String.IsNullOrEmpty(uploadFilename)) Then
                                '
                                ' -- version 5.0, separate class so this project can be built with contensive 5.0 reference, but run against contensive 4.1
                                Dim ErrorMessage As String = ""
                                Dim installDependencies As Boolean = Not cp.Doc.GetBoolean(rnBlockDependencies)
                                If (InstallController.installCollectionFromUpload(cp, rnUploadCollectionFile, ErrorMessage)) Then
                                    If (Not String.IsNullOrEmpty(ErrorMessage)) Then
                                        '
                                        ' -- install successful, but a problem
                                        form.body &= cp.Html.p("Installation completed with the follow message [" & ErrorMessage & "].")
                                    Else
                                        '
                                        ' -- install failed
                                        form.body &= cp.Html.p("Installation successful.")
                                    End If
                                Else
                                    form.body &= cp.Html.p("Error installing collection files, ERROR: " & ErrorMessage)
                                End If
                            End If
                        End If
                        '
                        ' Get Form
                        form.addRow()
                        form.rowName = "Collection Zip File"
                        form.rowValue = cp.Html.InputFile(rnUploadCollectionFile)
                        form.addRow()
                        form.rowName = "Block Dependencies"
                        form.rowValue = cp.Html.CheckBox(rnBlockDependencies)
                        form.addFormHidden("UploadCount", "1")
                        form.addFormButton(ButtonOK)
                        form.addFormButton(ButtonCancel)
                    End If
                    returnResult = form.getHtml(cp)
                End If
                Return returnResult
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
    End Class
End Namespace

File: source/addonManager51/Addons/UploadClass.cs
using System;
using Contensive.BaseClasses;
using Microsoft.VisualBasic.CompilerServices;

namespace Contensive.Addons.AddonManager51 {
    // 
    public class UploadClass : AddonBaseClass {
        // 
        // -- injected objects -- do not dispose
        private CPBaseClass cp;
        // 
        // -- class scope
        private const string RequestNameButton = "button";
        private const string ButtonCancel = " Cancel ";
        private const string ButtonOK = " OK ";
        // 
        // ====================================================================================================
        // 
        public override object Execute(CPBaseClass CP) {
            try {
                cp = CP;
                return getUpload();
            } catch (Exception ex) {
                CP.Site.ErrorReport(ex);
                throw;
            }
        }
        // 
        // ====================================================================================================
        // 
        private string getUpload() {
            string returnResult = "";
            try {
                string Button = cp.Doc.GetText(RequestNameButton);
                if ((Button ?? "") == ButtonCancel) {
                    // 
                    // ----- redirect back to the root
                    cp.Response.Redirect(cp.Site.GetText("adminUrl"));
                } else {
                    // 
                    // -- create form
                    var form = new PortalFramework.FormNameValueRowsClass() {
                        title = "Upload Collection",
                        body = cp.Html.p("Use this form to upload an add-on collection. If the GUID of the add-on matches one already installed on this server, it will be updated. If the GUID is new, it will be added."),
                        description = cp.Html.p("Upload a collection zip file to install the collection on this site. ")
                    };
                    if (!cp.User.IsAdmin) {
                        // 
                        // -- Put up error message
                        form.body += cp.Html.p("You must be an administrator to use this tool.");
                    } else {
                        if (Operators.CompareString(cp.Site.GetText("buildVersion"), cp.Version, false) < 0) {
                            // 
                            // -- database needs to be upgraded
                            form.description += cp.Html.p("WARNING: The site database needs to be upgraded. You should do this before installing addon collections.");
                        }
                        // 
                        // -- Upload tool
                        if ((Button ?? "") == ButtonOK) {
                            // 
                            // -- handle upload
                            if (cp.User.IsDeveloper & cp.Doc.GetBoolean("InstallCore")) {
                                // 
                                // -- Reinstall core collection
                                cp.Content.Delete("Add-on Collections", "ccguid='{8DAABAE6-8E45-4CEE-A42C-B02D180E799B}'");
                                // 
                                // -- use v5 method
                                string ErrorMessage = "";
                                if (!InstallController.installCollectionFromLibrary(cp, "{8DAABAE6-8E45-4CEE-A42C-B02D180E799B}", ref ErrorMessage)) {
                                    form.description += cp.Html.p("ERROR: " + ErrorMessage);
                                }
                            }
                            string uploadFilename = cp.Doc.GetText(constants.rnUploadCollectionFile);
                            if (!string.IsNullOrEmpty(uploadFilename)) {
                                // 
                                // -- version 5.0, separate class so this project can be built with contensive 5.0 reference, but run against contensive 4.1
                                string ErrorMessage = "";
                                bool installDependencies = !cp.Doc.GetBoolean(constants.rnBlockDependencies);
                                if (InstallController.installCollectionFromUpload(cp, constants.rnUploadCollectionFile, ref ErrorMessage)) {
                                    if (!string.IsNullOrEmpty(ErrorMessage)) {
                                        // 
                                        // -- install successful, but a problem
                                        form.body += cp.Html.p("Installation completed with the follow message [" + ErrorMessage + "].");
                                    } else {
                                        // 
                                        // -- install failed
                                        form.body += cp.Html.p("Installation successful.");
                                    }
                                } else {
                                    form.body += cp.Html.p("Error installing collection files, ERROR: " + ErrorMessage);
                                }
                            }
                        }
                        // 
                        // Get Form
                        form.addRow();
                        form.rowName = "Collection Zip File";
                        form.rowValue = cp.Html.InputFile(constants.rnUploadCollectionFile);
                        form.addRow();
                        form.rowName = "Block Dependencies";
                        form.rowValue = cp.Html.CheckBox(constants.rnBlockDependencies);
                        form.addFormHidden("UploadCount", "1");
                        form.addFormButton(ButtonOK);
                        form.addFormButton(ButtonCancel);
                    }
                    returnResult = form.getHtml(cp);
                }
                return returnResult;
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                throw;
            }
        }
    }
}